### PR TITLE
Add a hack for "+" in S3 URIs

### PR DESCRIPTION
**Pull request by dimaryaz**
Workaround for https://forums.aws.amazon.com/thread.jspa?threadID=55746

### CODE REVIEW

File: s3-proxy/s3.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/src/core/ngx_string.c b/src/core/ngx_string.c
+index 468e9600..7d8cfc9c 100644
```
Comment (akarve) [inline s3-proxy/s3.patch:2]: These first two lines shouldn't be here?

File: s3-proxy/Dockerfile [hunk 2]
```diff
@@ -14,12 +14,15 @@ RUN set -x \
...
+COPY s3.patch /root/s3.patch
+
 RUN set -x \
     && apt-get update \
-    && apt-get install -y gettext \
+    && apt-get install -y gettext patch \
     && apt-get install -y curl gcc make libssl-dev libpcre3-dev zlib1g-dev \
     && curl -L http://nginx.org/download/nginx-${NGINX_VERSION}.tar.gz | tar zx \
     && cd nginx-${NGINX_VERSION} \
+    && patch -p1 < /root/s3.patch \
```
Comment (akarve) [inline s3-proxy/Dockerfile:25]: minor: with this you don't need an further comments
```suggestion
    && patch -p1 < /root/s3_encode_plus.patch \
```

File: s3-proxy/s3.patch (hunk 1 above)
Comment (akarve) [reply s3-proxy/s3.patch:2]: Or do we need `git patch` if we retain these lines? Sorry, I've never seen this kind of patch, usually just `diff - u` output so not familiar.

Comment (dimaryaz) [reply s3-proxy/s3.patch:2]: Oh, that's just what `git diff` outputs; it's harmless.

But, didn't know about `git apply` - I don't need to install `patch`.

**Review by nl0 — APPROVED** [review]
ohwow looks magical

File: s3-proxy/s3.patch (hunk 1 above)
Comment (dimaryaz) [reply s3-proxy/s3.patch:2]: Oh wait, no, we don't have `git` here.

**Pull request closed without merging**